config/configtls/configtls.go

```diff
@@ -211,6 +221,19 @@ func (c Config) Validate() error {
 	return nil
 }
 
+func (c ServerConfig) Validate() error {
+	// For servers, both certificate and key are always required
+	if !c.hasCert() && !c.hasKey() {
+		return errors.New("TLS configuration must include both certificate and key for server connections")
+	}
```

Comment:
Very confused about the comment vs the code difference. You say both are required, but here you fail if both are missing.